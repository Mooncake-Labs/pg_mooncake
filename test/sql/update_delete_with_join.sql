CREATE TABLE s (a int, b int);
INSERT INTO s VALUES (1, 123), (1, 456);

CREATE TABLE t (c int, d int) USING columnstore;
INSERT INTO t VALUES (1, 0), (2, 0);

UPDATE t SET d = b FROM s WHERE c = a;
SELECT * FROM t ORDER BY c;

DELETE FROM t USING s WHERE c = a;
SELECT * FROM t ORDER BY c;

DROP TABLE s, t;
