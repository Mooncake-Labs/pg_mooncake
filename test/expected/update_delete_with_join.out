CREATE TABLE s (a int, b int);
INSERT INTO s VALUES (1, 123), (1, 456);
CREATE TABLE t (c int, d int) USING columnstore;
INSERT INTO t VALUES (1, 0), (2, 0);
UPDATE t SET d = b FROM s WHERE c = a;
SELECT * FROM t ORDER BY c;
 c |  d  
---+-----
 1 | 123
 2 |   0
(2 rows)

DELETE FROM t USING s WHERE c = a;
SELECT * FROM t ORDER BY c;
 c | d 
---+---
 2 | 0
(1 row)

DROP TABLE s, t;
